testgen/src/main/scala/TriangleTestGenerator.scala

```diff
@@ -0,0 +1,37 @@
+import java.io.File
+
+import testgen.TestSuiteBuilder._
+import testgen._
+
+object TriangleTestGenerator {
+  def main(args: Array[String]): Unit = {
+    val file = new File("src/main/resources/triangle.json")
+
+    def mapArgToString(arg: Any): String = {
+      arg match {
+        case vals: List[_] => vals.mkString(", ")
+        case _ => throw new IllegalStateException
+      }
+    }
+
+    def sutArgs(parseResult: CanonicalDataParser.ParseResult, argNames: String*): String =
+          argNames map (name => mapArgToString(parseResult(name))) mkString
+
+    def fromLabeledTest(argNames: String*): ToTestCaseData =
```

Comment:
I don't have a real strong opinion. But, I would be more comfortable with a stateless object if there was only a single method returning a type. But, in this case there are multiple methods potentially operating on the same values. And, a triangle can be multiple types..

At the end of the day - what is probably most important is what is the goal of the exercise? What is the exercise meaning to teach/re-enforce?  
It is always a bit difficult to figure out what a generic exercise is attempting to teach.